R/get-path-from-object.R

```diff
@@ -158,32 +158,188 @@ get_model_id.bbi_model <- function(.mod) {
 
 #' Get path to data file
 #'
-#' Helper to extract the path to the input data from a
-#' `bbi_{.model_type}_summary` or `bbi_{.model_type}_model` object. Note that
-#' **this function only works on models that have been successfully run**
-#' because the path to the data is constructed during model submission.
-#' @param .mod Model to use, either a `bbi_{.model_type}_model` or
-#' `bbi_{.model_type}_summary` object.
+#' Helper to extract the path to the input data from a `NONMEM` control stream file.
+#'
+#' @param .bbi_object The object to query. Could be
+#' a `bbi_{.model_type}_model` object,
+#' `bbi_{.model_type}_summary` object,
+#' or a tibble of class `bbi_log_df`.
+#' @param .check_exists If `TRUE`, the default, will throw an error if the file does not exist
+#' @param pull_from_config If `TRUE`, pull the data path from `bbi.json` created
+#'  during model submission, and compare to the path defined in the control stream.
 #' @param ... Arguments passed through to methods. (Currently none, but may have
 #'   some modifier arguments other model types in the future.)
-#' @importFrom fs path_rel
+#'
+#' @details
+#'
+#' ### Notes on model extensions:
+#' `bbi` has different handling for `.mod` vs `.ctl` extensions for `NONMEM`
+#' models. When a model such as `001.ctl` is targeted with `bbi`, `bbi` will
+#' create a new directory called `001`, copy the model into it (with a `.mod`
+#' extension), and do the execution directory. For more details, see `bbi`'s
+#' [documentation](https://github.com/metrumresearchgroup/bbi/blob/main/integration/markdown/expectations.md?plain=1#L19:L21)
```

Comment:
I don't see where this explains the difference between handling for `.mod` and `.ctl` files. It mentions bbi creating the output directory and copying the file into it, but that's the same in either case.

The difference is that on copy bbi adjusts a relative data path for `.mod` files, adding on a "../". Or, said from the user perspective, writers of `.ctl` files are expected to specify a relative path for the data file one level deeper than the control stream they're editing.

This motivation for the `.mod` adjustment was apparently inspired by the PsN behavior. (See [this](https://github.com/metrumresearchgroup/bbi/blob/835953efed10fff2d7024ac1c3dd88949b268c05/cmd/nonmem.go#L981-L984) bbi comment.)

Also, I wonder whether `submit_model()` is really the more useful place to document this handling (it can then be referenced from here). Unlike `get_data_path()`, that's an entry point we know a user will go through.

Reply:
Thanks for the added details. Will update the docs here to provide more context. However cc'ing @seth127 for the comment about where the documentation should live. My vote would actually be to include the info in both helper docs, but maybe that's a bit overkill?

Reply:
> My vote would actually be to include the info in both helper docs, but maybe that's a bit overkill?

In my view, yes, or rather it's an invitation for docs about something complicated to get out of sync with each other.  (If you felt strongly that you didn't want a reference here, you could use `@template` to avoid that.)